### PR TITLE
Conversion method from compressed matrix (csr, csc) to coordinate format (coo)

### PR DESCRIPTION
**Pull request by unnonouno**
I implemented csr2coo method.
Merge #234 first
related to #36

### CODE REVIEW

File: cupy/cusparse.py [hunk 1]
```diff
@@ -117,6 +117,60 @@ def cscsort(x):
...
+    indptr = cupy.empty(m + 1, 'i')
+    cusparse.xcoo2csr(
+        handle, x.row.data.ptr, x.nnz, m,
+        indptr.data.ptr, cusparse.CUSPARSE_INDEX_BASE_ZERO)
+    return cupy.sparse.csr.csr_matrix(
+        (x.data, x.col, indptr), shape=x.shape)
+
+
+def csr2coo(x, data, indices):
+    """Convert a CSR-matrix to COO format.
```
Comment (takagi) [inline cupy/cusparse.py]: Please begin with third-person singular.

File: cupy/cusparse.py [hunk 2]
```diff
@@ -117,6 +117,60 @@ def cscsort(x):
...
+
+    Args:
+        x (cupy.sparse.csr_matrix): A matrix to be converted.
+        data (cupy.ndarray): A data array for converted data.
+        indices (cupy.ndarray): A index array for converted data.
```
Comment (takagi) [inline cupy/cusparse.py]: "An index array"

File: cupy/cusparse.py [hunk 3]
```diff
@@ -117,6 +117,60 @@ def cscsort(x):
...
+
+    Returns:
+        cupy.sparse.coo_matrix: Returns a converted matrix.
```
Comment (takagi) [inline cupy/cusparse.py]: Please describe as a noun.

File: cupy/cusparse.py [hunk 4]
```diff
@@ -117,6 +117,60 @@ def cscsort(x):
...
+
+    """
+    handle = device.get_cusparse_handle()
+    m = x.shape[0]
+    nnz = len(x.data)
+    row = cupy.empty(nnz, 'i')
```
Comment (takagi) [inline cupy/cusparse.py]: `numpy.intp` would be better.

Comment (unnonouno) [inline cupy/cusparse.py]: It must be int32: http://docs.nvidia.com/cuda/cusparse/index.html#cusparse-lt-t-gt-csr2coo

**Comment by unnonouno** [comment]
I updated it.

File: cupy/cusparse.py (hunk 4 above)
Comment (takagi) [inline cupy/cusparse.py]: Okay.

**Comment by takagi** [comment]
LGTM!

**Pull request merged**